pkg/operator/ceph/cluster/osd/update.go

```diff
@@ -142,6 +144,7 @@ func (c *updateConfig) updateExistingOSDs(errs *provisionErrors) {
 			errs.addError("failed to update OSD %d. failed to extract OSD info from existing deployment %q. %v", osdID, depName, err)
 			continue
 		}
+		c.osdDesiredState[osdID] = &osdInfo
```

Comment:
what will happen if we update the device class here itself???

and we can just add spec `osddfoutput` instead of `osdDesiredState` to `updateconfig`??

Reply:
> what will happen if we update the device class here itself???

My proposal is an optimization as described [here](https://github.com/rook/rook/pull/14457/files#r1679851313).

> and we can just add spec osddfoutput instead of osdDesiredState to updateconfig??

The osd df output is the actual state, which we only query at the end of the reconcile.

Reply:
>The osd df output is the actual state, which we only query at the end of the reconcile.

Yaa I think that make sense, but before update and after the update the deviclass values  willremains the same,

But the approach would be useful if we need to utilize this for other changes like resizing.